### PR TITLE
Stats: Support "start_date" to query chart data

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/red-team/issues/267

## Proposed Changes

* Support querying API endpoint `stats/visits` with `start_date` for chart data.

## Why are these changes being made?

* Query `stats/visits` with `start_date` to get partial period data as a part of this project.

## Testing Instructions

* Spin this change up with the Calypso Live branch.
* Apply date range with the feature flag to the URL for the `week` period: `/stats/week/{site-slug}?flags=stats/new-date-filtering&chartStart=2024-10-10&chartEnd=2024-10-17`.
* Ensure the chart bars display responding to the selected date range.

|Before|After|
|-|-|
|截圖 2024-11-18 凌晨12 17 18|截圖 2024-11-18 凌晨12 15 29|

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-chart-tabs/index.jsx [hunk 1]
```diff
@@ -145,14 +145,17 @@ const NO_SITE_STATE = {
...
-const memoizedQuery = memoizeLast( ( chartTab, date, period, quantity, siteId ) => ( {
-	chartTab,
-	date,
-	period,
-	quantity,
-	siteId,
-	statFields: QUERY_FIELDS,
-} ) );
+const memoizedQuery = memoizeLast(
+	( chartTab, date, period, quantity, siteId, startdate = '' ) => ( {
```
Comment (a8ck3n) [inline client/my-sites/stats/stats-chart-tabs/index.jsx]: The new param should probably be `startDate` right?

File: client/state/data-layer/wpcom/sites/stats/visits/index.js [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const fetch = ( action ) => {
 				query: {
 					unit: period,
 					date,
+					start_date,
```
Comment (a8ck3n) [inline client/state/data-layer/wpcom/sites/stats/visits/index.js]: Similar to how `stat_fields` is handled, should we do `start_date: startDate` for consistency here?

**Review by a8ck3n — APPROVED** [review]
This tests well for me. Just a few small comments on the var naming.

File: client/state/data-layer/wpcom/sites/stats/visits/index.js (hunk 2 above)
Comment (kangzj) [reply client/state/data-layer/wpcom/sites/stats/visits/index.js]: `startDate` is also used for the date selected. I wonder if we should either use `start_date` or `chartStart` to tell them apart 🤔

Comment (dognose24) [reply client/state/data-layer/wpcom/sites/stats/visits/index.js]: `chartStart` looks good to me!

Comment (dognose24) [reply client/state/data-layer/wpcom/sites/stats/visits/index.js]: Addressed in [aae53f8](https://github.com/Automattic/wp-calypso/pull/96480/commits/aae53f88746e9325dfa9b10ada4603c24f8f37e6).

**Pull request merged**